### ISSUE TITLE
XCM regression tests

### ISSUE THREAD

**Issue by xlc** [paritytech/polkadot-sdk#3097]
We had too many breaking XCM changes / bugs.
To ensure good compatibility with **existing** XCM usages, we should have a regression tests suite.

Steps:
- Create a script to collect top 20 unique XCM from live onchain data
- For each usage, write a test for it, could be using xcm-simulator or emulator or chopsticks
- Include the tests in CI
- Rerun the script every 3 months to identify if we have new type of XCM usage and write test for it

**Comment by bkchr** [comment]
https://github.com/polkadot-fellows/runtimes/issues/171 is probably the successor of this issue.

**Comment by Polkadot-Forum** [comment]
This issue has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-ecosystem-tests/9067/1

**Comment by rockbmb** [comment]
Hello all,

What's the status on this issue? Does it still make sense to pick this up?

I do not mean to be impudent; with [AcalaNetwork/e2e-tests](https://github.com/AcalaNetwork/e2e-tests) and [open-web3-stack/polkadot-ecosystem-tests](https://github.com/open-web3-stack/polkadot-ecosystem-tests) having good coverage, I was considering where this issue fit.

**Comment by acatangiu** [comment]
Yes, this is still actual. https://github.com/open-web3-stack/polkadot-ecosystem-tests gives us the framework but is currently missing the actual tests. Would be great if you could start adding tests there.

**Comment by acatangiu** [comment]
To remove ambiguity, we need list of desired tests. That way, we can tell how much is covered and not, etc.

Should also contain info about what is covered by PET, what is covered by emulated tests.

**Comment by xlc** [comment]
> Create a script to collect top 20 unique XCM from live onchain data

This is the key element of this issue so we know what's needed
